Gateway/Http/Client/BaseTransaction.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Adyen\Payment\Gateway\Http\Client;
+
+use Adyen\Payment\Helper\Data;
+use Adyen\Payment\Gateway\Request\Header\HeaderDataBuilder;
+use Magento\Payment\Gateway\Http\TransferInterface;
+use Magento\Payment\Gateway\Http\ClientInterface;
+
+abstract class BaseTransaction implements ClientInterface
+{
+
+    /**
+     * @var Data
+     */
+    protected Data $adyenHelper;
+
+    /**
+     * BaseTransactionClient constructor.
+     * @param Data $adyenHelper
+     */
+    public function __construct(Data $adyenHelper)
+    {
+        $this->adyenHelper = $adyenHelper;
+    }
+
+    /**
+     * Builds the headers if they are not provided.
+     *
+     * @param TransferInterface $transferObject
+     * @return array
+     */
+    protected function requestHeaders(TransferInterface $transferObject): array
+    {
+        $headers = $transferObject->getHeaders();
+
+        // If no headers exist, build them using HeaderDataBuilder
```

Comment:
If any of the request builders add header `H`, this method will not be called. Thus, this will result in not adding desired headers. I would recommend adding header data builder to command chain of the transaction clients in `di.xml`. Let's have a catch-up if this is not clear.